core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -0,0 +1,1018 @@
+              html: '<p>Dummy Feedback</p>'
+            }
+          },
+          id: 'TextInput',
+          customization_args: {
+            rows: {
+              value: 1
+            },
+            placeholder: {
+              value: {
+                unicode_str: '',
+                content_id: 'ca_placeholder_0'
+              }
+            }
+          },
+          confirmed_unclassified_answers: [],
+          hints: [
+            {
+              hint_content: {
+                content_id: 'hint_1',
+                html: '<p>This is a hint.</p>'
+              }
+            }
+          ],
+          solution: {
+            correct_answer: 'Solution',
+            explanation: {
+              content_id: 'solution',
+              html: '<p>This is a solution.</p>'
+            },
+            answer_is_exclusive: false
+          }
+        },
+        linked_skill_id: null,
+        card_is_checkpoint: true,
+        recorded_voiceovers: {
+          voiceovers_mapping: {
+            1: {},
+            ca_placeholder_0: {},
+            feedback_id: {},
+            solution: {},
+            hint_1: {}
+          }
+        }
+      },
+      question_state_data_schema_version: 45,
+      language_code: 'en',
+      version: 1,
+      linked_skill_ids: [],
+      inapplicable_skill_misconception_ids: [],
+    }];
+  });
+
+  beforeEach(() => {
+    alertsService = TestBed.inject(AlertsService);
+    answerClassificationService = TestBed.inject(AnswerClassificationService);
+    contextService = TestBed.inject(ContextService);
+    expressionInterpolationService =
+      TestBed.inject(ExpressionInterpolationService);
+    questionObjectFactory = TestBed.inject(QuestionObjectFactory);
+    readOnlyExplorationBackendApiService = TestBed.inject(
+      ReadOnlyExplorationBackendApiService);
+    questionPlayerEngineService = TestBed.inject(QuestionPlayerEngineService);
+    urlService = TestBed.inject(UrlService);
+  });
+
+  it('should load questions on initialization', () => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    let quesitonPlayerSpy =
+      spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    let explorationIdSpy =
+      spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    let questionPlayerModeSpy =
+      spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    let versionSpy =
+      spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    let readOnlySpy =
+      spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+        .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+
+    expect(quesitonPlayerSpy).toHaveBeenCalled();
```

Comment:
I'm not sure how the exploration thing works here.
@kevintab95 PTAL.

Reply:
Please don't just say "I'm not sure". For anything like this, create a [short debugging doc](https://github.com/oppia/oppia/wiki/Debugging-Docs) with all your thoughts/investigation and then we'll have something concrete to discuss.

Reply:
We always set Question player mode to true when initialized, `if (!this.questionPlayerMode) ` this check is redundant here and the block will never execute.
PTAL @kevintab95 (I could use some help here thanks!)

Reply:
Hi @Radesh-kumar, I'm serious about creating a debugging doc according to the guidelines in the wiki. (I already discussed this with @kevintab95 too.)

If you are asking for help, it is important to provide full context and explanation of what you've thought through so that someone can read it and give you a pointer in the right direction. It will also help you ask a more specific question and explain the limits of what you've investigated. Little throwaway tidbits like your two recent comments are less helpful and make it harder for someone to answer your question.

Reply:
PTAL @seanlip  https://docs.google.com/document/d/1MTdaf5GCZl6uCE3bDjO_NM8gd4rp3mH4flNy_qnqzQw/edit#
Done!